apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -1951,8 +1956,11 @@ def _start_bucket_grad_sync(self, buckets: List[GradientBucket]) -> None:
                     bucket.grads_shard = bucket.grads_shard.clone()
 
         # Side stream for communication
+        # If new bucket is ready before last bucket communication finishes, use multiple
+        # communication streams could help pipeline reduce-scatter and all-reduce.
         main_stream = torch.cuda.current_stream()
-        comm_stream = self._pipeline_streams[-1]
+        self._last_comm_stream_id = (self._last_comm_stream_id + 1) % len(self._comm_streams)
```

Comment:
nit-picking (for a follow-up)
```suggestion
        self._last_comm_stream_id = (self._last_comm_stream_id + 1) % self.pipeline_size
```